### ISSUE TITLE
PHP 8.4 Support

### ISSUE THREAD

**Issue by kbond** [zenstruck/foundry#687]

**Comment by nikophil** [comment]
I think this is still not possible:
```
  Problem 1
    - doctrine/mongodb-odm-bundle[5.0.0-RC1, ..., 5.1.x-dev] require doctrine/mongodb-odm ^2.6 -> satisfiable by doctrine/mongodb-odm[2.6.0, ..., 2.9.x-dev].
    - doctrine/mongodb-odm-bundle[4.6.0, ..., 4.7.x-dev] require doctrine/mongodb-odm ^2.3 -> satisfiable by doctrine/mongodb-odm[2.3.0, ..., 2.9.x-dev].
    - doctrine/mongodb-odm[2.3.0, ..., 2.9.x-dev] require friendsofphp/proxy-manager-lts ^1.0 -> satisfiable by friendsofphp/proxy-manager-lts[v1.0.0, ..., 1.x-dev].
    - friendsofphp/proxy-manager-lts v1.0.0 requires laminas/laminas-code ^3.4.1 -> satisfiable by laminas/laminas-code[3.4.1, ..., 3.5.x-dev].
    - friendsofphp/proxy-manager-lts v1.0.1 requires laminas/laminas-code ^3.4.1|^4.0 -> satisfiable by laminas/laminas-code[3.4.1, ..., 3.5.x-dev, 4.0.0, ..., 4.15.x-dev].
    - friendsofphp/proxy-manager-lts[v1.0.2, ..., 1.x-dev] require laminas/laminas-code ~3.4.1|^4.0 -> satisfiable by laminas/laminas-code[3.4.1, 3.4.x-dev (alias of dev-master), 4.0.0, ..., 4.15.x-dev].
    - laminas/laminas-code 3.5.x-dev is an alias of laminas/laminas-code dev-develop and thus requires it to be installed too.
    - laminas/laminas-code 3.4.x-dev is an alias of laminas/laminas-code dev-master and thus requires it to be installed too.
    - laminas/laminas-code 4.0.x-dev is an alias of laminas/laminas-code dev-dev-4.0 and thus requires it to be installed too.
    - laminas/laminas-code[dev-master, dev-develop, dev-dev-4.0, 3.4.1, ..., 3.5.x-dev] require php ^7.1 -> your php version (8.4.0alpha4) does not satisfy that requirement.
    - laminas/laminas-code[3.5.0, ..., 3.5.x-dev] require php ^7.3 || ~8.0.0 -> your php version (8.4.0alpha4) does not satisfy that requirement.
    - laminas/laminas-code[4.0.0, ..., 4.4.x-dev] require php ^7.4 || ~8.0.0 -> your php version (8.4.0alpha4) does not satisfy that requirement.
    - laminas/laminas-code[4.5.0, ..., 4.7.x-dev] require php >=7.4, <8.2 -> your php version (8.4.0alpha4) does not satisfy that requirement.
    - laminas/laminas-code[4.8.0, ..., 4.12.x-dev] require php ~8.1.0 || ~8.2.0 -> your php version (8.4.0alpha4) does not satisfy that requirement.
    - laminas/laminas-code[4.13.0, ..., 4.15.x-dev] require php ~8.1.0 || ~8.2.0 || ~8.3.0 -> your php version (8.4.0alpha4) does not satisfy that requirement.
    - Root composer.json requires doctrine/mongodb-odm-bundle ^4.6|^5.0 -> satisfiable by doctrine/mongodb-odm-bundle[4.6.0, ..., 4.7.x-dev, 5.0.0-RC1, ..., 5.1.x-dev].
```

TL;DR: we require (dev) `doctrine/mongodb-odm-bundle` which undeer the hoods requires `laminas/laminas-code` which has currently no version (even a dev version) which supports PHP 8.4

Beside of this, `php8.4-mongodb` is still not released

**Comment by nikophil** [comment]
waiting for https://github.com/laminas/laminas-code/pull/201
